src/MLJGLMInterface.jl

```diff
@@ -115,19 +125,125 @@ function split_X_offset(X, offsetcol::Symbol)
     return newX, _to_vector(offset)
 end
 
+# If `estimates_dispersion_param` returns `false` then the dispersion
+# parameter isn't estimated from data but known apriori to be `1`.
+estimates_dispersion_param(::LinearRegressor) = true
+estimates_dispersion_param(::LinearBinaryClassifier) = false
+
+function estimates_dispersion_param(model::LinearCountRegressor)
+    return GLM.dispersion_parameter(model.distribution)
+end
+
+function _throw_sample_size_error(model, est_dispersion_param)
+    requires_info = _requires_info(model, est_dispersion_param)
+
+    if isnothing(model.offsetcol)
+        offset_info = " `offsetcol == nothing`"
+    else
+        offset_info = " `offsetcol !== nothing`"
+    end
+
+    modelname = nameof(typeof(model))
+    if model isa LinearCountRegressor    
+        distribution_info = "and `distribution = $(nameof(typeof(model.distribution)))()`"
+    else
+        distribution_info = "\b"
+    end
+
+    throw(
+        ArgumentError(
+            " `$(modelname)` with `fit_intercept = $(model.fit_intercept)`,"*
+            "$(offset_info) $(distribution_info) requires $(requires_info)"
+        )
+    )
+    return nothing
+end
+
+""" 
+    _requires_info(model, est_dispersion_param)
+    
+Returns one of the following strings
+- "`n_samples >= n_features`", "`n_samples > n_features`"
+- "`n_samples >= n_features - 1`",  "`n_samples > n_features - 1`"
+- "`n_samples >= n_features + 1`", "`n_samples > n_features + 1`"
+"""
+function _requires_info(model, est_dispersion_param)
+    inequality = est_dispersion_param ? ">" : ">="
+    int_num = model.fit_intercept - !isnothing(model.offsetcol)
```

Comment:
@rikhuijzer what of the following?
```suggestion
    if model.fit_intercept
        int_num = isnothing(model.offsetcol) ? 1 : 0
    else
        int_num = isnothing(model.offsetcol) ? 0 : -1
    end
```

Reply:
Sorry! I was wrong. The original was probably better. Maybe a comment like "can be -1, 0, 1" but it wasn’t too bad on second thought